src/diffusers/image_processor.py

```diff
@@ -165,17 +172,37 @@ def preprocess(
 
     def postprocess(
         self,
-        image,
+        image: torch.FloatTensor,
         output_type: str = "pil",
-    ):
-        if isinstance(image, torch.Tensor) and output_type == "pt":
+        do_normalize: Optional[Union[List[bool], bool]] = None,
+    ):  
+        if not isinstance(image, torch.Tensor):
+            raise ValueError(
+                f"Input for postprocess is in incorrect format: {type(image)}.  we only support pytorch tensor"
+            )
+        if output_type not in ["latent", "pt", "np", "pil"]:
+            deprecation_message = (
+                f"the output_type {output_type} is outdated and has been set to `np`. Please make sure to set it to one of these instead: "
+                "`pil`, `np`, `pt`, `latent`"
+            )
+            deprecate("Unsupported output_type", "1.0.0", deprecation_message, standard_warn=False)
+            output_type = "np"
+
+        if output_type == "latent":
+            return image
+
+        if not isinstance(do_normalize, list):
+            do_normalize = image.shape[0] * [do_normalize or self.config.do_normalize]
```

Comment:
I assume the goal is to default to self.config.do_normalize _only_ when the argument is `None`, When the argument is False, this will default to `self.config.do_normalize`.

When we use optional boolean arguments in python, a good rule of thumb is to as soon as possible in the function set them to a default value by explicitly using `is None` and if there is other behavior based off of them being not set, set alternative flags for those.

i.e.

```
def foo(bar: Optional[bool]=None):
    if bar is None:
        bar = False # the default value
        bar_was_set = False
   else:
      bar_was_set = True

   # from here on, bar can always be treated as just a boolean and if anything
   # is needed based on if not being set, `bar_was_set` can be used
```